sql/src/main/java/io/crate/operation/scalar/string/StringUnaryScalar.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.string;
+
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.operation.Input;
+import org.apache.lucene.util.BytesRef;
+
+import java.util.function.Function;
+
+
+/**
+ * Scalar implementation that wraps another function: f(ByteRef) -> R
+ * <br />
+ * null values will result in null as output
+ */
+class StringUnaryScalar<R> extends Scalar<R, BytesRef> {
```

Comment:
Could this be changed to a UnaryScalar that takes the inputType as generic as well?